indra/llcharacter/llik.cpp

```diff
@@ -1407,6 +1426,46 @@ void LLIK::Joint::setLocalRot(const LLQuaternion& new_local_rot)
     }
 }
 
+// only call this if you know what you're doing
+// this should only be called once before starting IK algorithm iterations
+void LLIK::Joint::setLocalScale(const LLVector3& scale)
+{
+    // compute final scale adustment to applly to mLocalPos and mBone
+    constexpr F32 MIN_INVERTABLE_SCALE = 1.0e-15f;
+    LLVector3 re_scale;
+    for (S32 i = 0; i < 3; ++i)
+    {
+        // verify mLocalScale component to avoid introducing NaN
+        re_scale[i] = (mLocalScale[i] > MIN_INVERTABLE_SCALE) ?  scale[i] / mLocalScale[i] : 0.0f;
+    }
+    // We remember the final scale adjustment for later...
+    mLocalScale = scale;
```

Comment:
should this log a warning or check if mLocalScale == scale?   is that worth checking?

Reply:
No, I don't think we need a WARNING.  Eventually when we implement the "config once --> lasts forever" behavior we can add code that checks for non-changing config parameters.